File: solid_SRP.py
#Single Responsibility Principle

class Funcionario:
    def __init__(self, nome, salario):
        self.nome = nome
        self.salario = salario
    
    def calcular_salario(self):
        # Cálculo de salário (simples)
        return self.salario * 1.1  # aumento de 10%

class FuncionarioRepository:
    def salvar_no_banco(self, funcionario):
        # Lógica para salvar no banco de dados (exemplo simples)
        print(f"Salvando {funcionario.nome} com salário {funcionario.salario} no banco de dados.")

File: prova03/solid_SRP.js
// Classe com responsabilidade única
class Funcionario {
    constructor(nome, salario) {
      this.nome = nome;
      this.salario = salario;
    }
  
    calcularSalario() {
      return this.salario * 1.1; 
    }
  }
  

  class FuncionarioRepository {
    salvarNoBanco(funcionario) {

      console.log(`Salvando ${funcionario.nome} com salário ${funcionario.salario} no banco de dados.`);
    }
  }
  
  const funcionario = new Funcionario("Carlos", 1000);
  console.log(funcionario.calcularSalario());
  
  const repo = new FuncionarioRepository();
  repo.salvarNoBanco(funcionario);
  